### PR TITLE
enable dev deploy

### LINKED ISSUE #74: ADD: Dev Env
Add the following:

- [x] firebase dev env
- [x] dev branch
- [x] dev deploy Github Action w/ firebase dev project
- [x] test
- [x] docs (readme)

### PR DESCRIPTION
**Pull request by strictly-scarce**
Closes #74 

Signed-off-by: Strictly-Scarce <strictly_scarce1@protonmail.com>

### CODE REVIEW

**Comment by strictly-scarce** [comment]
This also makes the following repo changes:

* Default branch is now `dev`
* Master is protected, merges must be reviewed

image

**Comment by yellowgh0st** [comment]
We need to preserver master as default branch because Vercel uses default production for production environment
and I can't change it at the moment.

File: .github/workflows/deploy-dev.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        with:
+          name: build
+          path: build
+      - name: Deploy to Firebase
+        uses: w9jds/firebase-action@master
+        with:
+          args: deploy --only hosting --project vether-dapp-dev
+        env:
+          FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
+          REACT_APP_INFURA_API: ${{ secrets.REACT_APP_INFURA_API }}
```
Comment (yellowgh0st) [inline .github/workflows/deploy-dev.yml:41]: We need to investigate why this is not working.
Firebase probably uses something like firebase.config.key to access them.

**Review by yellowgh0st — APPROVED** [review]

**Comment by yellowgh0st** [comment]
> This also makes the following repo changes:
> 
> * Default branch is now `dev`
> * Master is protected, merges must be reviewed

I changed this back due to reason I mentioned and merged the changes.

**Pull request merged**